### ISSUE TITLE
Security vulnerability - (hacked) Client sending incorrect type will cause crashes.

### ISSUE THREAD

**Issue by Xd-pro** [jojoe77777/FormAPI#75]
# Problem
When a client submits a form with a result of something that isn't applicable to the form type, the server will crash due to a type error if the plugin developer has tried to statically type the result in their closure in the constructor or `createFormType` methods.

# Example
```php
$form = new SimpleForm(function(Player $recipient, int /*this type is key*/ $result = null) {
  if ($result === null) {
    return true;
  }
  $recipient->sendMessage("Form button clicked!");
})

$form->addButton("A button");
$hackedClientUser->sendForm($form);
```

Now, a user with a hacked client designed to exploit this vulnerability will submit the form with a different result type, a string for example, then the closure will be called, causing a type error because a string was passed to the closure instead of an int.

# Soloution
In `Form->handleResponse`, check the type of `$data` is appropriate to the form type. If it isn't, make the result null or kick the player.

**Comment by Xd-pro** [comment]
#61 Will fix this if it is accepted.

**Comment by dktapps** [comment]
Fixed by #61
